app/controllers/session_controller.rb

```diff
@@ -10,13 +10,19 @@ def new
 
   #POST /session
   def create
-    @user = User.find_by(email: params[:email])
-    if @user && @user.authenticate(params[:password])
-      session[:current_user_id] = @user.id
-      # p @user.id
-      redirect_to user_path(@user)
+    if @user = User.find_by(email: params[:email])
+      if @user && @user.authenticate(params[:password])
```

Comment:
is there any way we can refactor this to avoid a nested if statement? feels like this method is doing a lot now

Reply:
create a private method called authenticate?